controllers/reconcilers/kserve_inferenceservice_reconciler.go

```diff
@@ -140,7 +144,11 @@ func (r *KserveInferenceServiceReconciler) DeleteKserveMetricsResourcesIfNoKserv
 
 	for i := len(inferenceServiceList.Items) - 1; i >= 0; i-- {
 		inferenceService := inferenceServiceList.Items[i]
-		if utils.IsDeploymentModeForIsvcModelMesh(&inferenceService) {
+		isvcDeploymentMode, err := utils.GetDeploymentModeForIsvc(ctx, r.client, &inferenceService)
+		if err != nil {
+			return err
+		}
+		if isvcDeploymentMode == utils.Serverless {
```

Comment:
Isn't the logic inverted here?
```suggestion
		if isvcDeploymentMode != utils.Serverless {
```